src/gr11review/part2/Utility.java

```diff
@@ -1,5 +1,621 @@
+*/
+  public static double avgWord(String filenametxt) throws IOException {
+    BufferedReader file = new BufferedReader(new FileReader(filenametxt));
+      //varaibles
+      String wordread;
+      String readingworld;
+      double finalnumber;
+      double average;
+      double intcount2;
+      //initializing varaibles
+      finalnumber = 0;
+      intcount2 = 0;
+      wordread = "";
+      readingworld="";
+      
+      
+      //while loop to read every word
+      while((wordread = file.readLine()) != null){
+        //wordread = file.readLine();
+       finalnumber = wordread.length() + finalnumber;
+        intcount2 = intcount2 +1;
+       
+
+       } 
+        file.close();
+        average = finalnumber/intcount2;
+        
+        return average;
+      }
+      
+        //either24
+        /**
+* an array of ints, return true if the array contains a 2 next to a 2 or a 4 next to a 4, but not both.
+*
+* @param nums gets the array and performs the either24 code
+* @return either is true or false if 2 is beside a 2 or a 4 beside 4
+* @author Addison Chan
+*/
+      
+      public static boolean either24(int[] nums){
+    //variables
+    int intcount3;
+    boolean inttwos = false;
+    boolean intfours = false;
+  
+      for(intcount3 = 0; intcount3<nums.length -1; intcount3++){
+        if(nums[intcount3]==2 && nums[intcount3+1]==2)inttwos = true;
+        if(nums[intcount3]==4 && nums[intcount3]==4)intfours = true;
+      
+    }
+      return(inttwos ^ intfours);
+    }
+    
+      //maxSpan
+      /**
+* the leftmost and righmost appearances of some value in an array. We'll say that the "span" is the 
+* number of elements between the two inclusive. A single value has a span of 1. Returns the largest span 
+* found in the given array. (Efficiency is not a priority.)
+*
+* @param nums gets the array and performs the max span of the array
+* @return finds the span of the array
+* @author Addison Chan
+*/
+      public static int maxSpan(int[] nums){
+        //variables
+      int intcount4;
+      int intcount5;
+      int intspan;
+      int intcalculation;
+      //initializing 
+      intspan = 0;
+        for(intcount4 = 0; intcount4 < nums.length; intcount4++){
+          for(intcount5 = 0; intcount5 < nums.length; intcount5++){
+            if(nums[intcount4] == nums[intcount5]){
+            intcalculation = intcount5 - intcount4 + 1;
+            intspan = Math.max(intcalculation,intspan);
+
+      }
+
+    }
+    }
+    return intspan;
+    
+  }
+  //year average
+  /**
+* a method that defines a two dimensional array that holds the table of scoring average data shown above 
+* (the grey area). The method will compute the average value of each of the 5 years and output the 
+* averages to a file yearAvg.txt. Each value will be on it's own line.
+*
+* @param N/A
+* @return prints the 6 averages into the txt document
+* @author Addison Chan
+*/
+  public static void yearAvg() throws IOException {
```

Comment:
@addisonc1 Based on the specified numbers in the provided table for this problem, the outputted average values are not correct
```
6.8636718750000005
7.788920593261719
8.712066096067428
8.340672133187763
8.710705750520265
```